multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,330 @@
+            corresponds to the centroids of view 1 and centroids[1] corresponds
+            to the centroids of view 2. These are not yet the final cluster
+            centroids.
+        '''
+
+        # Compute consensus vectors for final clustering
+        v1_consensus = list()
+        v2_consensus = list()
+
+        for clust in range(self.n_clusters):
+            v1_distances = cdist(Xs[0], centroids[0])
+            v1_partitions = np.argmin(v1_distances, axis=1).flatten()
+            v2_distances = cdist(Xs[1], centroids[1])
+            v2_partitions = np.argmin(v2_distances, axis=1).flatten()
+
+            # Find data points in the same partition in both views
+            part_indices = (v1_partitions == clust) * (v2_partitions == clust)
+
+            # Recompute centroids based on these data points
+            if (np.sum(part_indices) != 0):
+                cent1 = np.mean(Xs[0][part_indices], axis=0)
+                v1_consensus.append(cent1)
+
+                cent2 = np.mean(Xs[1][part_indices], axis=0)
+                v2_consensus.append(cent2)
+
+        # Check if there are no consensus vectors
+        self.centroids_ = [None, None]
+        if (len(v1_consensus) == 0):
+            msg = 'No distinct cluster centroids have been found.'
+            raise ConvergenceWarning(msg)
+        else:
+            self.centroids_[0] = np.vstack(v1_consensus)
+            self.centroids_[1] = np.vstack(v2_consensus)
+
+            # Check if the number of consensus clusters is less than n_clusters
+            if (self.centroids_[0].shape[0] < self.n_clusters):
+                msg = ('Number of distinct cluster centroids ('
+                       + str(self.centroids_[0].shape[0])
+                       + ') found is smaller than n_clusters ('
+                       + str(self.n_clusters)
+                       + ').')
+                raise ConvergenceWarning(msg)
+
+            # Updates k if number of consensus clusters less than original
+            # n_clusters value
+            self.n_clusters = self.centroids_[0].shape[0]
+
+    def fit(self, Xs):
+
+        '''
+        Fit the cluster centroids to the data.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            This list must be of size 2, corresponding to the two views of
+            the data. The two views can each have a different number of
+            features, but they must have the same number of samples.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        '''
+
+        Xs = check_Xs(Xs, enforce_views=2)
+
+        # Run multi-view kmeans for n_init different centroid initializations
+        min_inertia = np.inf
+        best_centroids = None
+
+        for _ in range(self.n_init):
+
+            # Random initialization of centroids
```

Comment:
just so you know, this is a huge disadvantage for your alg vs. initializing with k-means++ and in my experience matters a lot. not going to make you look into it now, but I'd make an issue at least

Reply:
also, consider in the future adding option for custom initialization

Reply:
Will do